pyrad/packet.py

```diff
@@ -92,12 +92,14 @@ def __init__(self, code=0, id=None, secret=six.b(''), authenticator=None,
             raise TypeError('authenticator must be a binary string')
         self.authenticator = authenticator
         self.message_authenticator = None
+        self.raw_packet = None
 
         if 'dict' in attributes:
             self.dict = attributes['dict']
 
         if 'packet' in attributes:
-            self.DecodePacket(attributes['packet'])
+            self.raw_packet = attributes['packet']
+            self.DecodePacket(self.raw_packet)
```

Comment:
All subclasses of `Packet` were checking for the `packet` attribute and setting `self.raw_packet` if it was present, so I moved that to the base class.